### PR TITLE
Set background colour on required field validation

### PR DESCRIPTION
**Pull request by jschultz**
This pull request highlights empty required fields on validation by setting their background colour to red.

### CODE REVIEW

File: gooey/gui/model.py [hunk 1]
```diff
@@ -207,8 +208,18 @@ def skipping_config(self):
...
+    error_found = False
+    for arg in self.required_args:
+      widget = arg.widget_instance.widget_pack.widget
+      if arg.value:
+        widget.SetBackgroundColour(wx.NullColour)
+      else:
+        if not error_found:
+          error_found = True
+          widget.SetFocus()
+        widget.SetBackgroundColour("Red")
```
Comment (chriskiehl) [inline gooey/gui/model.py]: I think straight red is a little too bold for Gooey's muted palette. Additionally, you keep the font color the same, so it all end up looking a little wonky. 

Could you soften it up a bit? `#EF9A9A` looked pretty good to my eye.

File: gooey/gui/model.py [hunk 2]
```diff
@@ -207,8 +208,18 @@ def skipping_config(self):
     return self.build_spec['manual_start']
 
   def is_required_section_complete(self):
-    completed_values = filter(None, [arg.value for arg in self.required_args])
-    return len(self.required_args) == len(completed_values)
+    error_found = False
+    for arg in self.required_args:
+      widget = arg.widget_instance.widget_pack.widget
```
Comment (chriskiehl) [inline gooey/gui/model.py]: You should call `widget.Refresh()` on everything. When there are multiple required fields, I'm  only seeing the first one change colors until I mouse over the rest.

File: gooey/gui/model.py [hunk 3]
```diff
@@ -207,8 +208,18 @@ def skipping_config(self):
     return self.build_spec['manual_start']
 
   def is_required_section_complete(self):
-    completed_values = filter(None, [arg.value for arg in self.required_args])
```
Comment (chriskiehl) [inline gooey/gui/model.py]: Technically... _technically..._ you shouldn't be making UI changes here in the model and instead use the Presenter class, but, due to my many, many code organization sins, you'd probably have to change a whole lot of stuff in order to make that work. Further, all this code all changes in 2.0, so ultimately it fine living here

File: gooey/gui/model.py (hunk 1 above)
Comment (jschultz) [reply gooey/gui/model.py]: Done

File: gooey/gui/model.py (hunk 2 above)
Comment (jschultz) [reply gooey/gui/model.py]: I can't reproduce this. But I've done as you requested, if you can check that it works as intended.

**Comment by jschultz** [comment]
I'm afraid my git skills leave something to be desired. I tried following instructions to rebase, and now it tells me that I'm trying to merge 27 commits?!

**Comment by bilderbuchi** [comment]
What are you trying to accomplish and what did you try?

**Comment by chriskiehl** [comment]
No pain quite like git pain. : ) 

The good news is that git works _really_ hard to not lose your work. Without knowing exactly how you got to the current state, I'd wager that the safest bet is approach this in three steps. 1. do a little reflog spelunking and branch off of your last known good commit to get yourself back in a good state,  (2) clean up your history, (3) bring yourself up to date with master.  

**Recovering your original state:**

It'll look something like this: 

`git reflog`

That'll show you a chronological list of all your git activity. Along the lines of: 

```
a26df8f HEAD@{0}: checkout: Set background colour on required field validation
...
```

You should be able to find your last known good commit before you started rebasing (probably easiest to identify by commit message rather than sha). Once you found that, all you got to do is check it out 

```
git checkout [YOUR_SHA]
```

Then, create a new branch from that point 

```
git checkout -b new-branch-name 
```

Once you done that, a `git log` should show you all the history you expect. 

**2. Fix your local history:**

The easiest way to do this is to not rebase at all, but instead `reset --soft` back to when you started your work. 

```git log --oneline```

it'll print out a list of commits

```
a26df8f you commit 1
bebc05b your commit 2
a7a6421 your commit 3
...
1b2c2e1 a commit from master
```
You'll want to reset soft back to that **first commit that isn't yours**

```
git reset --soft SHA_OF_THE_FIRST_COMMIT_THAT_ISNT_YOURS
```

That'll undo all your commits and put everything back into git staging, which you can verify with 

```
git status
```

You should see all your changes staged. I'd do a `git diff` at this point to verify!

If all looks correct, you can just commit as usual with a nice message. 

```
git commit -m "your feature" 
```


**3. Bringing yourself up to date with master:**

Checkout master and pull

```
git checkout master
git pull
``` 

Then hop back to your branch 

```
git checkout YOUR_BRANCH_NAME
```

And finally, rebase _against_ master 

```
git rebase master
```

Something more or less like that ; )

**Comment by jschultz** [comment]
Thanks for your patience! After messing it up a few more times I finally made it work. Let's see how I go with those other pull requests now.

**Comment by chriskiehl** [comment]
👍

**Pull request merged**